### PR TITLE
Introduces Solr-based alternative for the Browse Lists feature

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #1626 by migrating away from an approach which relies exclusively on ActiveRecord and PostgreSQL for the browse list querying.

### CODE REVIEW

File: app/controllers/orangelight/browsables_controller.rb [hunk 1]
```diff
@@ -1,59 +1,179 @@
...
+    if browsing_names?
+      @facet = 'author_s'
+    elsif browsing_subjects?
+      @facet = 'subject_facet'
+    elsif browsing_titles?
+      @facet = 'name_title_browse_s'
+    end
+  end
+
+  # All of this needs to be restructured into a QueryService Class
```
Comment (jrgriffiniii) [inline app/controllers/orangelight/browsables_controller.rb]: I still stand by this, but I can split this work into a pull request if that is acceptable.

File: app/controllers/orangelight/browsables_controller.rb [hunk 2]
```diff
@@ -1,59 +1,179 @@
...
+    query_response["response"]["docs"].map { |doc| odm_class.new(doc) }
+  end
+
+  ## The queries
+
+  def solr_default_params
+    {
+      sort: solr_default_sort,
+      fl: solr_default_fields,
+      rows: 100_000_0
```
Comment (jrgriffiniii) [inline app/controllers/orangelight/browsables_controller.rb]: I was uncertain how to locate the constant for this. From what I understoo, Ruby does not really offer a solid ceiling for `Integer`.

File: app/controllers/orangelight/browsables_controller.rb [hunk 3]
```diff
@@ -1,59 +1,179 @@
...
+    solr_query = solr_model_query(model)
+    range_parameters = {
+      q: solr_query
+    }
+    range_parameters[:fq] = "index_i:[#{start} TO #{last}]" unless start.nil? && last.nil?
+
+    solr_default_params.merge(range_parameters)
+  end
+
+  def find_range_by_model(model:, start:, last:)
```
Comment (jrgriffiniii) [inline app/controllers/orangelight/browsables_controller.rb]: These queries should probably follow a `QueryService` pattern, with each query instead being offered as a method on this Class.

File: app/controllers/orangelight/browsables_controller.rb [hunk 4]
```diff
@@ -1,59 +1,179 @@
...
+
+  # Updates the results-per-page variable using the request parameters
+  def update_rpp
     # if rpp isn't specified default is 50
     # if rpp has values other than 10, 25, 50, 100 then set it to 50
     # previous/next page links need to pass
     # manually set rpp
 
-    @model = model_table_name
     if rpp_param.nil?
```
Comment (jrgriffiniii) [inline app/controllers/orangelight/browsables_controller.rb]: Is it possible that some of this functionality might be ported into `kaminari`?

File: app/models/call_number_document.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def holding_id
+    value = fetch(:holding_id_s, nil)
+    Array.wrap(value).first
+  end
+
+  def location
+    value = fetch(:location_s, nil)
+    Array.wrap(value).first
```
Comment (jrgriffiniii) [inline app/models/call_number_document.rb]: These `Array.wrap` statements need to be refactored.

File: lib/orangelight/browse_lists.rb [hunk 6]
```diff
@@ -7,151 +7,248 @@
...
         faraday.options[:timeout] = 2000
-        faraday.request  :url_encoded             # form-encode POST params
-        faraday.response :logger                  # log requests to STDOUT
-        faraday.adapter  Faraday.default_adapter  # make requests with Net::HTTP
+        faraday.request(:url_encoded)
+        faraday.response(:logger)
+        faraday.adapter(Faraday.default_adapter)
       end
-      [sql_command, facet_request, conn]
     end
```
Comment (jrgriffiniii) [inline lib/orangelight/browse_lists.rb]: This should be replaced with `Blacklight.default_index.connection`, as this wraps the same client.

File: lib/tasks/browse.rake [hunk 7]
```diff
@@ -3,115 +3,73 @@
...
 namespace :browse do
-  desc 'Pull data for names browse'
-  task :names do
-    sql_command, facet_request, conn = BrowseLists.connection
-    BrowseLists.browse_facet(sql_command, facet_request, conn,
-                             'author_s', 'orangelight_names')
+  desc 'Run the development Solr'
+  task solr: [:environment] do
+    run_browse_lists_solr(Rails.env, port: '8983', persist: true) do
+      # Populate the browse list collection
```
Comment (jrgriffiniii) [inline lib/tasks/browse.rake]: Remove this comment.

**Comment by jrgriffiniii** [comment]
Blocks https://github.com/pulibrary/orangelight/issues/1903

**Comment by jrgriffiniii** [comment]
This is now passing with sufficient test coverage, however I will first proceed with deriving some additional issues from the improvements identified in the review of these proposed changes.

File: app/controllers/orangelight/browsables_controller.rb (hunk 3 above)
Comment (jrgriffiniii) [reply app/controllers/orangelight/browsables_controller.rb]: I have begun implementing this on https://github.com/pulibrary/orangelight/tree/issues-1626-jrgriffiniii-browse-lists-query-service

**Comment by christinach** [comment]
closing as stale

**Pull request closed without merging**